tools/setup-buildtools.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
```

Comment:
This script does a lot. I would prefer a README, rather than running this on my workstation.

Reply:
I'll make it optional. It bootstraps a lot of common C++ goodies, so it's easier to write a docker file using that script.

Reply:
...although I think for Linux containers - it'd be better to install a lot of these as a docker build image dependencies, so that the build script doesn't have to install again. Perhaps we should have a separate commit with docker configs for Windows and several common Linux distros.

Reply:
I agree that automating (and thereby documenting) build deps would be very useful wrt Docker.

I'm happy to keep this script here, I'm just wary of running things myself (especially under sudo) ;)

Reply:
This was essentially the minimum set of libs + tools required to compile some our own telemetry SDK stack, to bootstrap on any default image, such as Raspberry Pi / Raspbian, or an older Ubuntu 14.xx, 16.xx, RedHat, etc.. It did accumulate some meat over time. But in essence it's something that can be just scheduled to run inside of a GitHub Action on machine which by default may not have all the needed deps. It doesn't require docker. It runs once and creates **.buildtools** file, indicating that the build tools have been deployed, so it won't rerun again for incremental builds.

Alternate solution is to create separate set of docker image configs (quite a bit of duplication). Then assuming that each image is cooked properly with the right set of tools- it'd be then as easy as invoking cmake in that image, not needing the setup of all build tools.

Most big projects in some way would still enforce their own tooling, their own compiler, etc. like **chromium/tools/depot_tools** and would still be much more bulky than this shell script. I can truncate that to one command running apt-get install -qq all_deps_at_once + share some docker configs for an equivalent set of tools.

Reply:
I'm happy to keep the script. Let's see what others think.